bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -531,26 +561,40 @@ public void handleBookiesThatJoined(Set<BookieSocketAddress> joinedBookies) {
         return networkLocs;
     }
 
+    /*
+     * this method should be called in readlock scope of 'rwLock'
+     */
+    protected Set<BookieSocketAddress> addDefaultZoneBookiesIfMinNumRacksIsEnforced(
+            Set<BookieSocketAddress> excludeBookies) {
+        Set<BookieSocketAddress> comprehensiveExclusionBookiesSet;
+        if (enforceMinNumRacksPerWriteQuorum) {
+            comprehensiveExclusionBookiesSet = new HashSet<BookieSocketAddress>(excludeBookies);
+            topology.getLeaves(getDefaultRack()).forEach((Node node) -> {
+                if (node instanceof BookieNode) {
+                    BookieNode bookieNode = (BookieNode) node;
+                    comprehensiveExclusionBookiesSet.add(bookieNode.getAddr());
+                }
```

Comment:
Else?

Reply:
yeah we should handle `else` branch.

Reply:
I think I added this if block just to be on safe side. Leaves of "getDefaultRack()" should be BookieNode. In no normal circumstances it should be non-BookieNode (InnerNode). Anyhow do you have any strong suggestion for what needs to be in else block?

@eolivelli @sijie

Reply:
No I was only pointing here, maybe something that I missed was missing.

Reply:
@reddycharan  if we don't expect anything otherthan bookie node, we should just remove the if condition. Or add an else with exception. It really doesn't make sense to leave it like that.

Reply:
will add error logging